### PR TITLE
Update readonly ref from master

### PR DESCRIPTION
**Pull request by OmarTawfik**
@VSadov @dotnet/roslyn-compiler Updating the feature branch with master commits.

### CODE REVIEW

**Comment by OmarTawfik** [comment]
@dotnet-bot test this please

**Comment by OmarTawfik** [comment]
@dotnet-bot test this please

**Comment by OmarTawfik** [comment]
@VSadov these are the bits that I adjusted after the merge:
https://github.com/dotnet/roslyn/pull/17784/commits/631f6e4f9eee9c8063650a83f839c137a84bb2b9
https://github.com/dotnet/roslyn/pull/17784/commits/7661c7249c931af0bc616aa9f92be1f92308c6be

File: src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs [hunk 1]
```diff
@@ -1980,7 +1987,7 @@ internal bool CheckValueKind(BoundExpression expr, BindValueKind valueKind, Diag
                     return CheckEventValueKind((BoundEventAccess)expr, valueKind, diagnostics);
                 case BoundKind.DynamicMemberAccess:
                 case BoundKind.DynamicIndexerAccess:
-                    return true;
+                    return CheckDynamicValueKind(expr, valueKind, diagnostics);
```
Comment (VSadov) [inline src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs]: I was concerned about this change, but conflicting change has not been  merged in to the branch yet, so we should be fine.

**Pull request merged**